### PR TITLE
Add 2nu ECEC generator

### PR DESCRIPTION
**Pull request by gonzaponte**
This PR is a resurrection of #26, which was closed due to repository configuration issues.

This PR adds a generator for 2-neutrino double electron capture.
Example macros have also been added for reference and testing.

### CODE REVIEW

File: source/generators/ECECGenerator.cc [hunk 1]
```diff
@@ -0,0 +1,158 @@
+// ----------------------------------------------------------------------------
+// nexus | ECECGenerator.h
```
Comment (paolafer) [inline source/generators/ECECGenerator.cc]: typo: `.h` --> `.cc`

**Comment by gonzaponte** [comment]
For reference, these are the hit energy spectra obtained with the example macro.

image
image

File: macros/NEW_2nuECEC.config.mac [hunk 2]
```diff
@@ -0,0 +1,23 @@
+### --------------------------------------------------------
+###  This init macro simulates bb0nu decays
```
Comment (paolafer) [inline macros/NEW_2nuECEC.config.mac]: Update this brief description both in the `init` and in the `config` file.

File: macros/NEW_2nuECEC.config.mac [hunk 3]
```diff
@@ -0,0 +1,23 @@
+### --------------------------------------------------------
+###  This init macro simulates 2-neutrino ECEC decays
```
Comment (paolafer) [inline macros/NEW_2nuECEC.config.mac]: I know I'm a bit of a pain, but could you use the same format as the other example macros :-)?

Comment (gonzaponte) [reply macros/NEW_2nuECEC.config.mac]: You can't beat me at that ;). Do you mean like the one I just pushed?

File: macros/NEW_2nuECEC.config.mac [hunk 4]
```diff
@@ -0,0 +1,27 @@
+## ----------------------------------------------------------------------------
+## nexus | NEW_2nuECEC.init.mac
```
Comment (paolafer) [inline macros/NEW_2nuECEC.config.mac]: Perfect! Just a typo: `init` --> `config` here.

Comment (gonzaponte) [reply macros/NEW_2nuECEC.config.mac]: Classic copy-paste mistake :)

File: source/generators/ECECGenerator.cc [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+
+ECECGenerator::~ECECGenerator()
+{
+  delete msg_;
+  delete shell_;
+  delete atom_;
+}
+
+
+G4AtomicShellEnumerator ECECGenerator::GetShellID(G4String s)
```
Comment (paolafer) [inline source/generators/ECECGenerator.cc]: I get a compiler warning because of the variable name: `warning: declaration shadows a variable in namespace 'CLHEP'`. Could you use another name instead of `s`? Anything that is not related to units of measure.

File: source/generators/ECECGenerator.cc [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+REGISTER_CLASS(ECECGenerator, G4VPrimaryGenerator)
+
+ECECGenerator::ECECGenerator():
+  G4VPrimaryGenerator(),
+  atomic_number_(-1),
+  shell_name_("K"), // K, L1, L2, L3, M1, ..., M5
+  region_("ACTIVE"),
+  msg_ (nullptr),
+
```
Comment (paolafer) [inline source/generators/ECECGenerator.cc]: Remove blank line.

File: source/generators/ECECGenerator.cc [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+  if (!detconst)
+    G4Exception("[ECECGenerator]", "ECECGenerator()", FatalException,
+                "Unable to load geometry.");
+
+  geom_ = detconst->GetGeometry();
+
+  atom_ = new G4UAtomicDeexcitation();
+  atom_->SetFluo (true);
+  atom_->SetAuger(true);
+  atom_->SetCutForSecondaryPhotons(0);
```
Comment (paolafer) [inline source/generators/ECECGenerator.cc]: Here the class is initialized with two class members (`minGammaEnergy` and `minElectronEnergy`) to zero. However, when you generate the particles in line 149, you comment that 0 can't be used and those variable are modified inside the function. Is there any reason why you should not set `0.1 eV` here?

Comment (gonzaponte) [reply source/generators/ECECGenerator.cc]: I think this was just the first thing I tried and then I realized that I needed to set it to >0 in the GenerateParticles function. At the time I didn't know that GenerateParticles overwrites these values. So probably these two lines can even be removed. What do you prefer?

Comment (paolafer) [reply source/generators/ECECGenerator.cc]: For what I can see in the code, those thresholds are used exclusively in the `GenerateParticles()` function, therefore it should be exactly the same; on top of that, that function requires to be passed those arguments. The whole structure looks quite confusing to me, but it's Geant4... I would remove the setters at the beginning, since they don't do anything.

Comment (gonzaponte) [reply source/generators/ECECGenerator.cc]: As discussed offline, these setters have no effect as the `GenerateParticle` method overrides them. They have been removed.

**Review by paolafer — APPROVED** [review]
This PR adds the code for the double electron capture generator, used in the simulations of the NEXT paper on ECEC. It was long due and it's finally approved!
